cupy_backends/cuda/libs/cublas.pyx

```diff
@@ -454,6 +454,8 @@ cdef _setStream(intptr_t handle):
     """
     if stream_module.enable_current_stream:
         setStream(handle, stream_module.get_current_stream_ptr())
+    elif stream_module.is_ptds_enabled():
+        setStream(handle, runtime.cudaStreamPerThread)
```

Comment:
Are you suggesting we replace those two lines simply by the following?
```python
setStream(handle, stream_module.get_current_stream_ptr())
```

Reply:
Yes, but it's already done above.

Reply:
It isn't, note how above there's `if stream_module.enable_current_stream`. The `enable_current_stream` is only set when using an explicit stream. However, we could perhaps remove that conditional too, I'm not sure if that would work though, my understanding is that `get_current_stream_ptr` returns `self.current_stream`, which is undefined until setting a stream, or am I missing something else?

Reply:
Ah, that I know. `self.current_stream` is a cdef attribute, so it's implicitly initialized to `0`. We should probably add a `__cinit__` method in `cupy_backends/cuda/stream.pyx` to ensure it's properly initialized (to whatever we eventually agree upon in https://github.com/cupy/cupy/pull/4322#discussion_r530601255. 

btw I would like to push a house keeping PR after this one is merged, so that everything in `cupy_backends/cuda/stream.pyx` is moved to `cupy/cuda/stream.pyx` (in the exactly opposite direction as proposed in https://github.com/cupy/cupy/pull/3584).

Reply:
For `stream_module.enable_current_stream`, I think @takagi is more knowledgable since he recently worked on the fix #4173. My understanding is `stream_module.enable_current_stream` is always set to `True` when a user attempt to use a stream (either as a context manager (`with stream:`) or explicitly turning it on (`stream.use()`).

However, I do think we should explicitly initialize it to `True`: note that it's a module attribute left uninitialized:
https://github.com/cupy/cupy/blob/d8f8a31393f5c7d3993cce951fcd29886b02c458/cupy_backends/cuda/stream.pxd#L3
So in a sense #4173 is a brute force fix (by overwriting it) 😅

Reply:
> However, I do think we should explicitly initialize it to `True`: note that it's a module attribute left uninitialized:

So when would it be `False`? If we're forcing an initialization to `True`, it becomes a completely unnecessary attribute, as I do see it being set to `False` anywhere.

Reply:
I feel it is completely unnecessary to have.

Reply:
I removed it, adjusted the `setStream` functions and `get_current_stream_ptr`. I'm not very happy with the condition in https://github.com/cupy/cupy/pull/4322/files#diff-c7ca9ae40b6014a65e9d4420604c0e83adb8aadd77011ce04fd2026d74f43a01R59-R60 , it doesn't look nice to compare to `<void*>0`, and I'm not sure what is the most appropriate/preferred/cleanest way to do that with Cython.

Reply:
`current_stream` could be changed to `intptr_t` IMHO so that we just compare with `0`. I don't have strong opinion on this, but would like to note this is done in one of the two `stream` modules, and best to be unified when they are merged.

Reply:
I did this now in https://github.com/cupy/cupy/pull/4322/commits/ece5c5569d65633861e4c8b139e791cedb2e2b6b, seems that this change doesn't require modifications elsewhere.